client/lib/preferences-helper/preference-list.js

```diff
@@ -14,41 +13,55 @@ import { getAllRemotePreferences } from 'calypso/state/preferences/selectors';
 import QueryPreferences from 'calypso/components/data/query-preferences';
 import Preference from './preference';
 
-class PreferenceList extends Component {
-	render() {
-		const { preferences, translate } = this.props;
-		return (
-			<div>
-				<QueryPreferences />
-				<a
-					href={ '/devdocs/client/state/preferences/README.md' }
-					title={ translate( 'Preferences' ) }
-				>
-					{ translate( 'Preferences' ) }
-				</a>
-				<Card className="preferences-helper__current-preferences">
-					{ ! isEmpty( preferences ) ? (
-						Object.keys( preferences ).map( ( preferenceName ) => (
-							<Preference
-								key={ preferenceName }
-								name={ preferenceName }
-								value={ preferences[ preferenceName ] }
-							/>
-						) )
-					) : (
-						<h5 className="preferences-helper__preference-header">
-							{ translate( 'No Preferences' ) }
-						</h5>
-					) }
-				</Card>
-			</div>
-		);
-	}
-}
+const PreferenceList = () => {
+	const translate = useTranslate();
 
-export default connect(
-	( state ) => ( {
-		preferences: getAllRemotePreferences( state ),
-	} ),
-	null
-)( localize( PreferenceList ) );
+	const preferences = useSelector( getAllRemotePreferences );
+	const sortedPreferenceItems = useMemo( () => {
```

Comment:
Does the `useMemo` perform any useful optimization here? I think it doesn't. The `PreferenceList` component doesn't have any props and rerenders only when the `preferences` selector returns a new value. But then `sortedPreferenceItems` needs to be re-calculated, despite the `useMemo`. If the `preferences` value doesn't change, `PreferenceList` is never rerendered and the memoized value never has a chance to be used.